src/leafletwidget/widget.ts

```diff
@@ -80,28 +77,51 @@ class MapWidget extends Widget {
    */
   protected onUpdateRequest(msg: Message): void {
     this.title.text = this._context.path.split('/').pop();
-    let cm = this._context.contentsModel;
-    if (cm === null) {
+    if (!this.isAttached) {
       return;
     }
-    let content = JSON.parse(this._context.model.toString());
+    let contentString = this._context.model.toString();
+    if (contentString === this._geojsonString) {
+      return;
+    }
+
+    // we're attached to the DOM and have new layer content now
     if (this._geojsonLayer) {
       this._map.removeLayer(this._geojsonLayer);
     }
-    this._geojsonLayer = leaflet.geoJson(content, {
-      pointToLayer: function (feature, latlng) {
-          return leaflet.circleMarker(latlng);
-      }
-    });
-    this._map.addLayer(this._geojsonLayer);
-    this._map.fitBounds(this._geojsonLayer.getBounds());
+    this._geojsonString = contentString;
+    this._geojsonLayer = null;
+    if (contentString) {
+      let content = JSON.parse(contentString);
+      this._geojsonLayer = leaflet.geoJson(content, {
+        pointToLayer: function (feature, latlng) {
+            return leaflet.circleMarker(latlng);
+        }
+      });
+      this._map.addLayer(this._geojsonLayer);
+      this._map.fitBounds(this._geojsonLayer.getBounds());
+    }
   }
 
   /**
    * A message handler invoked on a 'resize' message.
    */
-  onResize() {
+  onResize(msg: ResizeMessage) {
+    // Since we know the size from the resize message, we manually
+    // define getSize() so that it does not have to do a DOM read.
+    this._map.getSize = () => {
```

Comment:
Sure, I can expose that. I'll open a PR on the mono repo and CC you. Keep in mind though, it may not be *exactly* the information you want. i.e. you may need to account for border/padding on your child map node.